rfcs/0028-nix-release-model.md

```diff
@@ -0,0 +1,158 @@
+---
+feature: nix-release-model
+start-date: 2018-03-28
+author: Shea Levy <shea@shealevy.com>
+co-authors: (find a buddy later to help our with the RFC)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+A model of when, what, and how to make new releases of Nix.
+
+# Motivation
+[motivation]: #motivation
+
+We want to set clear expectations for users, provide guidance for
+developers, and provide both groups the benefits of easy availability
+of new improvements.
+
+# Detailed design
+[design]: #detailed-design
+
+The proposed process contains a number of interrelated components.
+I've numbered them for ease of reference and discussion, not to
+indicate any meaningful ordering. Any aspect of this can be suspended
+at the security team's discretion when handling a vulnerability. Items
+marked with a * need technical work to implement, which needn't delay
+the rest of the process.
+
+1. Keep `master` as close to release readiness as possible. This
+   includes testing, documentation, changelog entries, etc. being
+   ready *before* merge.
+2. \*Use code review and automated checks, enforced through branch
+    protection, to ensure point 1 (and generally improve quality and
+    increase knowledge spread).
```

Comment:
This seems to dictate having a CI to test the PRs which I am very much in favour of

Reply:
Yep! Though as noted above I don't think we need to *wait* for that to adopt this, adopting this would more be seen as a commitment to move toward that.